plugin/rewrite/edns0.go

```diff
@@ -199,12 +202,22 @@ func (rule *edns0VariableRule) ruleData(state request.Request) ([]byte, error) {
 		return []byte(state.Proto()), nil
 	}
 
+	if metadata.Labels(ctx) != nil {
```

Comment:
don't think this check is needed

Reply:
This also needs to check this is a label, meaning *metadata* needs to export a `IsLabel` functions that checks the syntax (2 words with a slash as seperator)

Reply:
Yes the check is needed to make the difference between 
- "metadata" is enabled : an unknown {label} is resolved as "".
- "metadata" is not enabled : {label} cannot be resolved and we return an error (current behavior)

=> I can add a IsMetadataEnabled() along with an IsLabel()